src/server/game/AI/SmartScripts/SmartAI.cpp

```diff
@@ -650,9 +650,7 @@ void SmartAI::OnCharmed(bool apply)
 {
     GetScript()->ProcessEventsFor(SMART_EVENT_CHARMED, NULL, 0, 0, apply);
 
-    if (!apply && !me->IsInEvadeMode() && me->GetCharmerGUID())
-        if (Unit* charmer = ObjectAccessor::GetUnit(*me, me->GetCharmerGUID()))
-            AttackStart(charmer);
+    CreatureAI::OnCharmed(apply);
```

Comment:
With all due respect, any smartscript mob is currently a huge pile of steaming bugs when mind controlled. They actively fight player control, cast spells without player input, and move without player input.

If you weigh this against the questionable benefit of having smart scripts still trigger while controlled (how many creatures with elaborate smartscripts can you think of that can be targeted by such abilities), there should be no question as to which is infinitely preferable.

PS: Your fix does not work. Mind Control is a possess, and is by far the most broken type of charm for SmartAI.